src/simulator_core/simulation/plotting.py

```diff
@@ -0,0 +1,99 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Plot result on the map for debugging."""
+import esdl
+import plotly.express as px
+import geopandas as gpd
+import numpy as np
+from shapely.geometry import LineString
+from numpy.typing import NDArray
+
+
+class Plotting:
+    """Plotting plot the results of asset (currently pipe) for inspection."""
+
+    @staticmethod
+    def plot_map(assets: list) -> None:
+        """Static function to plot the simulation results to the map.
+
+        :param assets: list collection of assets
+        """
+        # prepare list
+        geometry = []
+        asset_id = []
+        asset_name = []
+        mass_flow = []
+        pressure_in = []
+        pressure_out = []
+        temperature_in = []
+        temperature_out = []
+
+        for asset in assets:
+            if isinstance(asset.geometry, esdl.esdl.Line):
+                coor_pipe = []
+                for point in asset.geometry.point:
+                    coor_pipe.append([point.lon, point.lat])
+
+                geometry.append(LineString(coor_pipe))
+
+                asset_name.append(asset.name)
+                asset_id.append(asset.asset_id)
+                mass_flow.append(asset.output[-1]['mass_flow'])
+                temperature_in.append(asset.output[-1]['temperature_supply'])
+                temperature_out.append(asset.output[-1]['temperature_return'])
+                pressure_in.append(asset.output[-1]['pressure_supply'])
+                pressure_out.append(asset.output[-1]['pressure_return'])
+
+        # create geopanda instance
+        data = {'asset_id': asset_id,
```

Comment:
Can't you make an empty dataframe with all the required columns and then append the data? Then you won't need to construct the lists and this dictionary.

Reply:
yes, that can be another implementation